crates/runtime/src/memory.rs

```diff
@@ -9,9 +9,38 @@ use std::cell::RefCell;
 use std::convert::TryFrom;
 use wasmtime_environ::{MemoryPlan, MemoryStyle, WASM_MAX_PAGES, WASM_PAGE_SIZE};
 
+/// A memory allocator
+pub trait Allocator: Send + Sync {
+    /// Create new LinearMemory
+    fn new_memory(&self, plan: &MemoryPlan) -> Result<Box<dyn LinearMemory>, String>;
+}
+
+/// A default memory allocator used by Wasmtime
+pub struct DefaultAllocator;
+
+impl Allocator for DefaultAllocator {
```

Comment:
The term "Allocator" feels a bit general here, would it be possible to tweak the naming here to indicate that this isn't like a malloc/free allocator but rather an allocator of linear memories?